CIP-0048/README.md

```diff
@@ -0,0 +1,249 @@
+- This would allow users of 721 metadata to increase or reduce the amount of data an NFT asset uses.
+
+# Motivation
+
+- Large token mints that duplicated data could be dramatically reduced in size by pointing to one transaction payload that contains a ‘boilerplate’ structure.
+- 16kB is the upper limit of data in each transaction but If a user wanted more there is no alternative than to store that data off-chain using an external service such as ipfs.
+- There is no current mechanism to reduce duplicated metadata
+- NFT assets are restricted by there own metadata scope, optional references would prevent this restriction
+- There is currently no way to reference on chain data stored in a different policy
+
+# Specification
+
+### **Inside the 721 metadata JSON**
+
+| reserved | description                                            | scope                     |
+| -------- | ------------------------------------------------------ | ------------------------- |
+| refs     | references tag, defines where to find the onchain data | Inside `<asset_name>` tag |
+| p        | contains raw data (on chain data).                     | Inside `<policy_id>` tag  |
+
+### **The `refs` tag**
+
+Contains the `name`, `mediaType`, `src` tags (and the optional `type` tag)
+
+- `name` (string) is the references name (similar usage to name in the current `files` tag)
+- `mediaType` (string) defines the mime type for the data referenced in `src`
+  - is a string
+- `src` (string array) is an **ordered** array of references.
+  - The order is that of which the data will be parsed. For example `["1","3","2"]` will result in payload `"1"` being the start of the data, followed by payload `"3"` and ending with payload `"2"`.
+
+### **Optional `type` tag for `refs`**
+
+The default CIP-48 behavior is to assume the assets parent `<policy_id>` contains all the payloads.
+
+Optionally we define `type` and two options `policy` and `txhash`. To allow for non-default behavior
+
+- `policy`
+  - Defines an external policy (instead of the default behavior of using the parent policy)
+  - example
+    ```json
+      "refs": [
+        {
+          "name": "...",
+          "mediaType": "...",
+          "type": {
+            "policy": "<some_external_policy>"
+          }
+        }
+      ]
+    ```
+- `txhash`
+  - Defines an array of transaction hashes (instead of the default behavior of using the parent policy)
+  - example
+    ```json
+      "refs": [
+        {
+          "name": "...",
+          "mediaType": "...",
+          "type": {
+            "txhash": ["<mint_txhash>", "<mint_txhash>"]
+          }
+        }
+      ]
+    ```
+  - `txhash` is faster than parsing all transactions in a given policy.
+  - `txhash` can dramatically reduce onchain data usage if duplicated data is referenced here _(however, the bytes required to define the reference would have to be less than the reference data itself)_
```

Comment:
If we presume that the majority of references could be declared at a top-level utilizing CIP-88, this is just some rough thoughts but perhaps we could expand the references "type" to optionally point to another on-chain pointer or to a direct payload.

```cbor
{
  48: {
    0: '1.0.0', ; Version?
    1: [ ; Array of references
      {
        "name": "referenceA", ; Name of the reference
        "type": "pointer | payload", ; Maybe an enum of accepted options?
        "payload": { ; Example of a direct payload reference
          "mediaType": "text/plain",
          "content": "My Cool NFT Project"
        },
        "payload": { ; Example of a tx pointer reference?
          "type": "tx",
          "content": "[txhash]"
        },
        "payload": { ; Example of a token point reference?
          "type": "token",
          "content": [(policy_id),(asset_id)]
        }
      }
    ]
  }
}
```